pandas/tests/reshape/test_melt.py

```diff
@@ -1086,3 +1086,35 @@ def test_warn_of_column_name_value(self):
         with tm.assert_produces_warning(FutureWarning):
             result = df.melt(id_vars="value")
             tm.assert_frame_equal(result, expected)
+
+    def test_missing_stubname(self):
+        # GH46044
+        df = DataFrame({"id": ["1", "2"], "a-1": [100, 200], "a-2": [300, 400]})
+        result = wide_to_long(
+            df,
+            stubnames=["a", "b"],
+            i="id",
+            j="num",
+            sep="-",
+        )
+        index = pd.Index(
+            [("1", 1), ("2", 1), ("1", 2), ("2", 2)],
+            name=("id", "num"),
+        )
+        expected = DataFrame(
+            {"a": [100, 200, 300, 400], "b": [np.nan] * 4},
+            index=index,
+        )
+        tm.assert_frame_equal(result, expected)
+        # test extension string array
+        df_str = df.astype({"id": "string"})
```

Comment:
Can this test just be parameterized over `"dtype": [object, "string"]`?

Reply:
Done.